src/Confluent.Kafka/Serialization/LongDeserializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+
+using System;
+
+namespace Confluent.Kafka.Serialization
+{
+    /// <summary>
+    /// Derializer for Int64 data
+    /// </summary>
+    public class LongDeserializer : IDeserializer<long>
+    {
+        /// <summary>
+        /// Deserialize data to Int64
+        /// </summary>
+        /// <param name="data">input data</param>
+        /// <returns>deserialize result</returns>
+        public long Deserialize(byte[] data)
+        {
+            if (data == null)
+            {
+                // in accordance with java client.
+                return default(long);
```

Comment:
yeah, `object` looses type safety, I think that option is out. `Nullable` is unfortunate for reasons mentioned elsewhere.

I still (currently) think it is preferable to provide both because I think it's beneficial to have enforcement of no `null` values in cases where this reflects the data / usage (which I think will be the case in many scenarios). Generally, stronger typing brings more clarity to what's going on - I quickly understand more about what's going on simply by noting which serializer is being used.